### DISCUSSION TITLE
@react-aria and server-side rendering

### DISCUSSION THREAD

**Discussion by shannonrothe** [adobe/react-spectrum#1892] [Q&A] [answered]
👋

I'm building a component library upon `@react-aria` and facing some trouble with SSR functionality, particularly around tooltips and the associated hooks.

The issue I face is that when I build my library I end up with references to `document` in the output – which obviously doesn't cooperate server-side. I have read the docs and wrapped my component in `SSRProvider`, but still no luck. Any ideas? 🙏

Thanks for the great work on the libraries you offer, they're really helpful.

**Comment by snowystinger** [comment, marked as answer]
What references to document are you ending up with? maybe we forgot to protect a few in tooltips?
Looks like it may be this reference? https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/tooltip/src/useTooltipTrigger.ts#L72
if so, could you open an issue instead of a discussion so we can make it as a bug and track it?
thank you!
